src/ClientFactory/SocketFactory.php

```diff
@@ -31,6 +31,6 @@ public function createClient(array $config = [])
             throw new \LogicException('To use the Socket client you need to install the "php-http/socket-client" package.');
         }
 
-        return new Client($this->messageFactory, $config);
+        return new Client($this->messageFactory, $config, $config);
```

Comment:
thanks for looking into this.

i think the actual bug here is in the socket client. the aim was to be BC with version 1, but deprecate passing that factory. the signature of version 1 was correct for how we instantiated the client: https://github.com/php-http/socket-client/blob/v1.4.1/src/Client.php#L52

version 2 should at https://github.com/php-http/socket-client/blob/47e68cade6b76e7fcaaf648048845d9dde0accf9/src/Client.php#L61 should use $config2, not $config which i don't see how it could be useful...

i suggest we fix the socket client to work as intended. we could additionally add a version constant into the socket client to allow the bundle to call the constructor in a future proof way and avoid the deprecation warning.

Reply:
@joelwurtz what do you think, would it make sense that way?